### DISCUSSION TITLE
Registering IScheduler with AspNetCore app and package

### DISCUSSION THREAD

**Discussion by mniyad** [quartznet/quartznet#1291] [Q&A] [answered]
Hi
I'm trying to inject IScheduler in my Blazor app.  The jobs all run fine so I know the scheduler works.  
What's the recommended way to register IScheduler? I'm using AspNetCore hosting package

I attempted to register IScheduler as a singleton like below, however the IScheduler that is injected in my controller has no jobs or groups or anything. It's empty.

```C#
_jobs = await _scheduler.GetJobKeys(GroupMatcher<JobKey>.AnyGroup());
```

Service registration
```C#
// ASP.NET Core hosting
services.AddQuartzServer(options =>
{
  // when shutting down we want jobs to complete gracefully
  options.WaitForJobsToComplete = true;
});

//do I need to do this?
services.AddSingleton<IScheduler>(_ => StdSchedulerFactory.GetDefaultScheduler().GetAwaiter().GetResult());
```

**Comment by lahma** [comment, marked as answer]
Because getting a scheduler can lead to async path it's recommended that you inject the ISchedulerFactory to dependant and call GetScheduler on it when you need the reference.

**Reply by mniyad** [reply]
Thanks @lahma. That's exactly what I need
